pkg/controller/kogitoinfra/infinispan/infinispan_reconciler.go

```diff
@@ -0,0 +1,133 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infinispan
+
+import (
+	"fmt"
+	infinispanv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// InfinispanResource implementation of KogitoInfraResource
+type InfinispanResource struct {
+}
+
+// GetWatchedObjects provide list of object that needs to be watched to maintain Infinispan kogitoInfra resource
+func GetWatchedObjects() []framework.WatchedObjects {
+	return []framework.WatchedObjects{
+		{
+			GroupVersion: infinispanv1.SchemeGroupVersion,
+			AddToScheme:  infinispanv1.AddToScheme,
+			Objects:      []runtime.Object{&infinispanv1.Infinispan{}},
+		},
+		{
+			Objects: []runtime.Object{&corev1.Secret{}},
+		},
+	}
+}
+
+// Reconcile reconcile Kogito infra object
+func (i *InfinispanResource) Reconcile(client *client.Client, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (requeue bool, resultErr error) {
+
+	var infinispanInstance *infinispanv1.Infinispan
+
+	// Step 1: check whether user has provided custom infinispan instance reference
+	isCustomReferenceProvided := len(instance.Spec.Resource.Name) > 0
+	if isCustomReferenceProvided {
+		log.Debugf("Custom infinispan instance reference is provided")
+		resourceName := instance.Spec.Resource.Name
+		resourceNameSpace := instance.Spec.Resource.Namespace
+
+		infinispanInstance, resultErr = loadDeployedInfinispanInstance(client, resourceName, resourceNameSpace)
+		if resultErr != nil {
+			return false, resultErr
+		}
+		if infinispanInstance == nil {
+			return false, fmt.Errorf("custom Infinispan instance(%s) not found in namespace %s", resourceName, resourceNameSpace)
+		}
+	} else {
+		// create/refer kogito-infinispan instance
+		log.Debugf("Custom infinispan instance reference is not provided")
+		// if resource name is not provided then Infinispan instance should be created with default name = kogito-infinispan
+		resourceName := InstanceName
+
+		// if resource name is not provided then Infinispan instance should be created with default name = kogito-infinispan
+		resourceNameSpace := instance.Namespace
```

Comment:
I think we can use the provided variables instead of creating temps.

Reply:
I had removed these temp